arch/arm/core/aarch32/cortex_m/fault.c

```diff
@@ -131,7 +131,7 @@ LOG_MODULE_DECLARE(os);
  */
 
 #if (CONFIG_FAULT_DUMP == 1)
-static void fault_show(const z_arch_esf_t *esf, int fault)
+static void fault_show(const struct __esf *esf, int fault)
```

Comment:
I believe (actually, I hope :) ) that all these argument signature changes won't be needed if we accept the convention that the z_arch_esf_t will always be __esf, and contain the additional info at the end.

Reply:
> I think I like the new approach, but I have one last request for investigation, which I think will make the patch look even more "light".
> that is, try to define the extended register info as a last field inside the _esf (document this properly) and solely use the z_arch_esf_t in fault.c.

thanks for the review @ioannisg, yes I think that would work. The only nuance I see with that approach is the extra data would only be valid in the `esf_copy` passed in z_arm_fatal_error. That is, the pointer returned from `get_esf` would not contain the info. If just documenting that and reusing the struct sounds better to you, happy to make the change.

> Should we schedule this enhancement for Zephyr v2.4.0 release? Merge window closes in about 10 days, so if you have the bandwidth to work on this patch, we should be able to have it for next release.

Sounds great, I'll aim to wrap up the final changes you are suggesting over the next day or so!